### PR TITLE
[ci-visibility] Add github job name to the extracted CI metadata 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
* Read `GITHUB_JOB` when running in github actions.
* Update fixtures. 

### Motivation
`GITHUB_JOB` is now available as an environment variable in github actions (see https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables for more info), so we want to read it and attach it as a span tag.

### CODE REVIEW

File: packages/dd-trace/test/plugins/util/ci-env/buddy.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     {
       "BUDDY": "true",
       "BUDDY_EXECUTION_BRANCH": "master",
-      "BUDDY_EXECUTION_ID": "12",
+      "BUDDY_EXECUTION_ID": "buddy-execution-id",
```
Comment (juan-fernandez) [inline packages/dd-trace/test/plugins/util/ci-env/buddy.json:6]: unrelated change, this is just for convenience on the generation side

**Review by Qard — APPROVED** [review]
